src/noir/codec/scale.h

```diff
@@ -47,6 +47,21 @@ datastream<Stream>& operator>>(datastream<Stream>& ds, uint256_t& v) {
   return ds;
 }
 
+template<typename Stream, enum_c E>
```

Comment:
There is no naming convention related to concepts, but what about `noir::enumeration` for this?

```suggestion
template<typename Stream, enumeration E>
```